lib/handler/mruby/middleware.c

```diff
@@ -398,19 +400,40 @@ static int skip_tracing(h2o_conn_t *conn)
     return 1;
 }
 
+static int handle_header_raw_key(h2o_mruby_shared_context_t *shared_ctx, h2o_iovec_t *raw_key, h2o_iovec_t value, void *_req)
+{
+    h2o_req_t *req = _req;
+    const h2o_token_t *token;
+
+    h2o_iovec_t name = convert_env_key_to_header_name(&req->pool, raw_key->base, raw_key->len, 0);
+    if (name.base == NULL)
+        return 0;
+
+    if ((token = h2o_lookup_token(name.base, name.len)) != NULL) {
+        value = h2o_strdup(&req->pool, value.base, value.len);
+        h2o_add_header(&req->pool, &req->headers, token, NULL, value.base, value.len);
+    } else {
+        value = h2o_strdup(&req->pool, value.base, value.len);
+        h2o_add_header_by_str(&req->pool, &req->headers, name.base, name.len, 0, NULL, value.base, value.len);
+    }
```

Comment:
Instead of doing "if...is_token...else", you can simply rely on `h2o_add_header_by_str` with its `maybe_token` argument set to 1?

Reply:
Thank you for the suggestion. I did it in https://github.com/h2o/h2o/pull/2256/commits/7d3e42f3c41ad0724058435c3ed926ae915b548d